dependency/requirements.txt

```diff
@@ -4,7 +4,7 @@ cachetools~=5.3
 click<9.0
 cookiecutter>=2.1.1, <3.0
 dynaconf>=3.1.2, <4.0
-fsspec>=2021.4, <2023.1.1
+fsspec>=2021.4, <2023.2
```

Comment:
Do we need to bump the lower bound? Would it work for the older version of `fsspec`?

Reply:
It doesn't work with the lower version, but my reasoning to not bound the lower version was because you only need this newer version if you use the tar/zip config. I think only a very small part of users will use that feature, so I don't think it's necessary to force everyone to a strict version of fsspec, especially if they're using other dependencies that need a certain fsspec version. 

I will add a comment to the release notes about needing the latest fsspec for this functionality.

Reply:
If you do a `pip install` in a clean environment it will install the latest version of `fsspec`. If someone already has `fsspec` installed and then installs kedro, it could be that they don't get the most recent version. Then they'd have to do `pip install -U kedro`. I've also tried pinning the upper bound and it does seem like it's fine with the other requirements so perhaps it doesn't matter if we bump it that much. It's definitely cleaner than adding in logic to detect the `fsspec` version only if you read config from a compressed file.

Reply:
So it turns out fsspec `2023.1.0` is the last version that supports python 3.7. All versions above have dropped 3.7 support and the same goes for the `gcfs` and `adlfs` packages, so I've added a separate pin for 3.7 and python above 3.7.